can/interfaces/socketcand/socketcand.py

```diff
@@ -13,12 +13,115 @@
 import socket
 import time
 import traceback
+import urllib.parse as urlparselib
+import xml.etree.ElementTree as ET
 from collections import deque
+from typing import List
 
 import can
 
 log = logging.getLogger(__name__)
 
+DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS = ""
+DEFAULT_SOCKETCAND_DISCOVERY_PORT = 42000
+
+
+def detect_beacon(timeout_ms) -> List[can.typechecking.AutoDetectedConfig]:
+    """
+    Detects socketcand servers
+
+        This is what :meth:`can.detect_available_configs` ends up calling
+        search for available socketcand servers with fixed timeout of 3.1
```

Comment:
```suggestion
        to search for available socketcand servers with fixed timeout of 3.1
```

Reply:
Just my opinion, but the mix of milliseconds in the code and seconds in the commentary could cause headaches and bugs for people in the future (and in the present 😎).  

I would suggest seriously think about whether it might be a good plan to work through top-to-bottom in this PR to achieve consistency.